core/domain/interaction_jobs_one_off.py

```diff
@@ -194,12 +194,14 @@ def map(item):
                 )
             except Exception as e:
                 error_messages.append(
-                    '%s: %s' % (state.interaction.id, python_utils.UNICODE(e)))
+                    '%s: %s'.encode(encoding='utf-8') % (
+                        state.interaction.id, e))
 
         if error_messages:
+            error_msg = ', '.encode(encoding='utf-8').join(error_messages)
```

Comment:
This seems like an odd place to put it -- you're only encoding the comma.

Typically we do all the encoding at the very end when we have the final string. Does that work? (Please make sure to test manually, with actual unicode data.)

Reply:
I checked it - it is working correct since I am also separately encoding error messages above this line. Though updated this to encode complete string instead of just the comma.